hook.js

```diff
@@ -149,6 +151,14 @@ function ensureArrayWithBareSpecifiersFileUrlsAndRegex (array, type) {
     throw new Error(`'${type}' option only supports bare specifiers, file URLs or regular expressions. Invalid entries: ${inspect(invalid)}`)
   }
 
+  // Rather than evaluate whether we have a node: scoped built-in-module for
+  // every call to resolve, we just add them to include/exclude now.
+  for (const each of array) {
+    if (typeof each === 'string' && !each.startsWith('node:') && builtinModules.includes(each)) {
```

Comment:
Let's update the docstring of `ensureArrayWithBareSpecifiersFileUrlsAndRegex` to indicate that we handle this `node:` prefixed case.